docs/user-guide/slicing.ipynb

```diff
@@ -217,6 +226,174 @@
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Value based slicing\n",
+    "\n",
+    "Data in a [Dataset](../generated/scipp.Dataset.rst#scipp.Dataset) or [DataArray](../generated/scipp.DataArray.rst#scipp.DataArray) can be sliced by the coordinate value.\n",
+    "Value indexing with a [Variable](../generated/scipp.Variable.rst#scipp.Variable)  or an [Variable](../generated/scipp.Variable.rst#scipp.Variable) range is made via `__getitem__` and `__setitem__` with a dimension label as first argument.\n",
+    "This is available for DataArrays and Datasets, as well as items of a Dataset, but not for Variable types as coordinates are necessary.\n",
+    "In all cases a *view* is returned, i.e., just like when slicing a [numpy.ndarray](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.html#numpy.ndarray) no copy is performed.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Value slicing on point coordinates\n",
+    "\n",
+    "Lets create a simple `DataArray`. It has dimension coordinates in x and y:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "da = sc.DataArray(data = sc.Variable(dims=['x', 'y'], values=np.arange(8.0).reshape(2, 4)),\n",
+    "    coords={\n",
+    "        'x': sc.Variable(['x'], values=np.arange(2.0), unit=sc.units.m),\n",
+    "        'y': sc.Variable(['y'], values=np.arange(4.0), unit=sc.units.m)})\n",
+    "sc.show(da)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lets say we wish to slice the `DataArray` where y coord is exactly at the point 1.0 meters expressed as a variable. You will notice that the data item in the output have no y coordinate dependency after the point slicing operation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "point_value = 0.0 * sc.units.m\n",
+    "da['y', point_value] "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our y-coordinate is represents point data rather than bin edges for the dataset data. This means that when slicing with a point value, the value needs to exactly match a coordinate value. If it does not, an `IndexError` is raised."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "point_value = 0.1 * sc.units.m # No y coordinate value at this point\n",
+    "try:\n",
+    "    d['y', point_value]\n",
+    "except IndexError as e:\n",
+    "    print(str(e))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also slice a range just like index slicing above. No exact match to coordinate value of start/stop is required in the case of range slicing."
```

Comment:
Should mention that this selects all the points falling *between* the start/stop, and also define precisely the behavior at the bounds (I think it was open on one side, but closed on the other?).